docs/user_guide/job.rst

```diff
@@ -110,6 +110,11 @@ folder with meta.json.
 There is only one server, and only one app can be deployed to it for the Job, so "server" can appear only once in
 the ``deploy_map``.
 
+The ``deploy_map`` cannot be empty, and "@all" when specified as a site name carries a special meaning of all
+sites to deploy to. If "@all" is used, there should be no other apps being deployed to the sites. If an empty list of
+sites is specified for an app in the ``deploy_map``, then that app is to be deployed to no sites, and no validation is
+done other than checking that the folder exists.
```

Comment:
should we add some examples here?
Allowed:

```
deploy_map: {
    "app1": ["@ALL"], "app2": []
}
```

Not allowed:

```
deploy_map: {
    "app1": ["@ALL"], "app2": ["site-1"]
}
```
Not allowed:
```
deploy_map: {}
```